common/eth2_config/src/lib.rs

```diff
@@ -73,40 +82,154 @@ impl<'a> Eth2NetArchiveAndDirectory<'a> {
-                genesis_is_known: $genesis_is_known,
-            };
-
-            // A wrapper around `std::include_bytes` which includes a file from a specific network
-            // directory. Used by upstream crates to import files at compile time.
-            #[macro_export]
-            macro_rules! $macro_title {
-                ($base_dir: tt, $filename: tt) => {
-                    include_bytes!(concat!(
-                        $base_dir,
-                        "/",
-                        predefined_networks_dir!(),
-                        "/",
-                        $name,
-                        "/",
-                        $filename
-                    ))
+/// Indicates that the `genesis.ssz.zip` file is present on the filesystem. This means that the
+/// deposit ceremony has concluded and the final genesis `BeaconState` is known.
+const GENESIS_STATE_IS_KNOWN: bool = true;
+
+#[derive(Copy, Clone, Debug, PartialEq)]
+pub struct HardcodedNet {
+    pub name: &'static str,
+    pub genesis_is_known: bool,
+    pub config: &'static [u8],
+    pub deploy_block: &'static [u8],
+    pub boot_enr: &'static [u8],
+    pub genesis_state_bytes: &'static [u8],
+}
+
+/// Defines an `Eth2NetArchiveAndDirectory` for some network.
+///
+/// It also defines a `include_<title>_bytes!` macro which provides a wrapper around
+/// `std::include_bytes`, allowing the inclusion of bytes from the specific testnet directory.
+macro_rules! define_archive {
+    ($name_ident: ident, $name_str: tt, $genesis_is_known: ident) => {
+        paste! {
+            #[macro_use]
+            pub mod $name_ident {
+                use super::*;
+
+                pub const ETH2_NET_DIR: Eth2NetArchiveAndDirectory = Eth2NetArchiveAndDirectory {
+                    name: $name_str,
+                    unique_id: $name_str,
+                    genesis_is_known: $genesis_is_known,
                 };
+
+                /// A wrapper around `std::include_bytes` which includes a file from a specific network
+                /// directory. Used by upstream crates to import files at compile time.
+                #[macro_export]
+                macro_rules! [<include_ $name_ident _file>] {
+                    ($this_crate: ident, $base_dir: tt, $filename: tt) => {
+                        include_bytes!(concat!(
+                            $base_dir,
+                            "/",
+                            $this_crate::predefined_networks_dir!(),
+                            "/",
+                            $name_str,
+                            "/",
+                            $filename
+                        ))
+                    };
+                }
             }
         }
     };
 }
 
-define_net!(pyrmont, include_pyrmont_file, "pyrmont", true);
+/// Creates a `HardcodedNet` definition for some network.
+#[macro_export]
+macro_rules! define_net {
+    ($this_crate: tt, $mod: ident, $include_file: tt) => {{
```

Comment:
```suggestion
    ($this_crate: ident, $mod: ident, $include_file: tt) => {{
```

Seems like this works as is, but should this be `ident`?